internal/pkg/consensus/testing.go

```diff
@@ -239,14 +242,14 @@ func SeedFirstWinnerInNRounds(t *testing.T, n int, ki *crypto.KeyInfo, networkPo
 	}
 }
 
-func winsAtEpoch(t *testing.T, epoch uint64, ticket block.Ticket, ki *crypto.KeyInfo, networkPower, numSectors, sectorSize uint64, sectorInfos ffi.SortedPublicSectorInfo) bool {
-	signer := types.NewMockSigner([]crypto.KeyInfo{*ki})
-	wAddr, err := ki.Address()
-	require.NoError(t, err)
+func winsAtEpoch(t *testing.T, epoch abi.ChainEpoch, ticket block.Ticket, miner address.Address, networkPower, numSectors, sectorSize uint64, sectorInfos ffi.SortedPublicSectorInfo) bool {
 	em := ElectionMachine{}
 
 	// use ticket to seed postRandomness
-	postRandomness, err := em.GeneratePoStRandomness(ticket, wAddr, signer, epoch)
+	rnd := crypto.FakeRandomnessSource{Ticket: ticket}
+	entropy, err := mining.MakeElectionPoStChallengeSeedEntropy(epoch, miner)
+	require.NoError(t, err)
+	postRandomness, err := rnd.Randomness(context.Background(), acrypto.DomainSeparationTag_ElectionPoStChallengeSeed, epoch, entropy)
 	require.NoError(t, err)
 
 	// does this postRandomness create a winner?
```

Comment:
The call to GenerateCandidates below consumes the postRandomness VRF proof.  Right now go-sectorbuilder interface says that it wants to take in an abi.PoStRandomness.  I’m not sure if that type name matches the spec like go-filecoin’s code so you’ll want to investigate to see if you actually want to draw randomness from the vrf proof you have here before passing in.

Reply:
I still need to investigate this.

Reply:
There's nothing in the Go side that takes the digest. Lotus also takes a digest (SHA :-( ) before invoking the sectorbuilder method. 

There are numerous other divergences in Lotus from the chain sampling and randomness generation code  I've written, FYI @sternhenri, though the general structures now match.

Reply:
Yikes. Let me look into it some more on lotus end.